### PR TITLE
app-crypt/veracrypt: new package

### PR DESCRIPTION
**Pull request by soredake**

### CODE REVIEW

**Comment by anthonyryan1** [comment]
That's a lot of patches for something brand new to the tree, would it be better to wait for the next release?

File: app-crypt/veracrypt/veracrypt-1.19.ebuild [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+inherit flag-o-matic linux-info multilib toolchain-funcs wxwidgets eutils pax-utils
```
Comment (anthonyryan1) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: Remove eutils, multilib and alphabetize the remaining inherits.

**Comment by mgorny** [comment]
@soredake, please address the CI errors first.

File: app-crypt/veracrypt/files/veracrypt.init [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/sbin/openrc-run
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+depend() {
+	after localmount
+	if [ -e /lib/librc.so ]; then
```
Comment (gktrk) [inline app-crypt/veracrypt/files/veracrypt.init]: Per https://www.gentoo.org/support/news-items/2012-02-14-baselayout-1-deprecation.html, baselayout-1 is long gone, we don't need this check here.

File: app-crypt/veracrypt/files/veracrypt.init [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		need device-mapper
+	fi
+}
+
+start() {
```
Comment (gktrk) [inline app-crypt/veracrypt/files/veracrypt.init]: Per https://www.gentoo.org/support/news-items/2012-02-14-baselayout-1-deprecation.html, baselayout-1 is long gone, we don't need this check here.

File: app-crypt/veracrypt/files/veracrypt.init [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	if [ ! -e /lib/librc.so ]; then
+		eerror "The ${myservice} init script is written for baselayout-2"
+		eerror "Please do not use it with baselayout-1"
+		return 1
+	fi
+}
+
+stop() {
```
Comment (gktrk) [inline app-crypt/veracrypt/files/veracrypt.init]: Per https://www.gentoo.org/support/news-items/2012-02-14-baselayout-1-deprecation.html, baselayout-1 is long gone, we don't need this check here.

File: app-crypt/veracrypt/veracrypt-1.19.ebuild [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	if use X; then
+		newicon Resources/Icons/VeraCrypt-48x48.xpm veracrypt.xpm
+		make_desktop_entry ${PN} "VeraCrypt" ${PN} "System"
+	fi
+
+	pax-mark -m "${D}/usr/bin/veracrypt"
+}
+
+pkg_postinst() {
+	elog "There is an init script for VeraCrypt for Baselayout-2."
```
Comment (gktrk) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: Per https://www.gentoo.org/support/news-items/2012-02-14-baselayout-1-deprecation.html, baselayout-1 is long gone, we need to adjust this message accordingly.

File: app-crypt/veracrypt/veracrypt-1.19.ebuild [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+	elog "If you are a baselayout-2 user and you would like the VeraCrypt"
+	elog "mappings removed on shutdown in order to prevent other file systems"
+	elog "from unmounting then run:"
+	elog "rc-update add veracrypt boot"
+	elog
+
+	ewarn "If you're getting errors about DISPLAY while using the terminal"
```
Comment (gktrk) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: I remember hitting this back in the day. Is this still a problem? I personally can't reproduce it.

File: app-crypt/veracrypt/veracrypt-1.19.ebuild [hunk 7]
```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+inherit flag-o-matic linux-info pax-utils toolchain-funcs wxwidgets
```
Comment (gktrk) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: `flag-o-matic` will not be needed after we remove the use of `append-flags`

Comment (gktrk) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: Per https://devmanual.gentoo.org/eclass-reference/eutils.eclass/index.html, the use of `newicon` requires you to inherit `eutils`

File: app-crypt/veracrypt/files/veracrypt.init [hunk 8]
```diff
@@ -0,0 +1,33 @@
+#!/sbin/openrc-run
+# Copyright 1999-2015 Gentoo Foundation
```
Comment (gktrk) [inline app-crypt/veracrypt/files/veracrypt.init]: The copyright year should be 2016

File: app-crypt/veracrypt/veracrypt-1.19.ebuild [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+	"${FILESDIR}/${PV}/execstack-fix.diff"
+	"${FILESDIR}/${PV}/link-with-libdl.patch"
+	"${FILESDIR}/${PV}/remove-packaging-from-makefile.patch"
+)
+
+src_prepare() {
+	if use X; then
+		# Fix linking when NOGUI=1
+		sed -e "s/WX_CONFIG_LIBS := base/&,core/" -i Main/Main.make || die "sed Main/Main.make failed"
+	fi
```
Comment (soredake) [inline app-crypt/veracrypt/veracrypt-1.19.ebuild]: Compiles for me too, seems this fix is not needed anymore.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/2e60672...4702d71 with modifications. Thanks!

**Pull request closed without merging**